hypervisor/arch/x86/guest/assign.c

```diff
@@ -398,9 +399,10 @@ static struct ptirq_remapping_info *add_intx_remapping(struct acrn_vm *vm, uint3
 		} else {
 			/* The mapping has already been added to the VM. No action
 			 * required. */
+			entry_is_updated = false;
 		}
 
-		if (entry != NULL) {
+		if (entry != NULL && entry_is_updated == true) {
```

Comment:
entry_is_updated  is already bool. No need to use "=="

Reply:
fixed on pr3223